### PR TITLE
Dedicated camb

### PR DESCRIPTION
**Pull request by drboldwin**
https://github.com/rpherbig/dr-scripts/issues/857

### CODE REVIEW

**Comment by drboldwin** [comment]
So the one drawback I foresee is that this is really only effective for dedicated_camb_use: spell

**Comment by drboldwin** [comment]
also does anything else call invoke or find_charge_invoke_stow other than crossing-training and combat-trainer?

File: combat-trainer.lic [hunk 1]
```diff
@@ -1654,6 +1654,9 @@ class GameState
 
     @analyze_combo_type = settings.analyze_combo_type
     echo("  @analyze_combo_type: #{@analyze_combo_type}") if $debug_mode_ct
+	
+	@dedicated_camb_use = settings.dedicated_camb_use
```
Comment (rpherbig) [inline combat-trainer.lic]: This identation looks off.

File: combat-trainer.lic [hunk 2]
```diff
@@ -1881,7 +1884,7 @@ class GameState
       echo('check_charging?: done') if $debug_mode_ct
 
       cambrinth_charges(nil)
-      invoke(@cambrinth)
+      invoke(@cambrinth, @dedicated_camb_use)
```
Comment (rpherbig) [inline combat-trainer.lic]: I think you also need to update line 823.

**Comment by rpherbig** [comment]
`buff.lic` uses `find_charge_invoke_stow`

`healme.lic` uses `charge_and_invoke`

**Comment by drboldwin** [comment]
kk

File: combat-trainer.lic (hunk 2 above)
Comment (drboldwin) [inline combat-trainer.lic]: Hmm I think this is more complicated than just adding it to 823.  That is sending your ritual focus into invoke - I'm not sure what would happen if I try to invoke inkwell spell

File: combat-trainer.lic (hunk 1 above)
Comment (drboldwin) [inline combat-trainer.lic]: fixed - stupid tabs

File: combat-trainer.lic (hunk 2 above)
Comment (rpherbig) [inline combat-trainer.lic]: Good point. But you'll still have to pass in something or Ruby will complain about the number of arguments.

Comment (drboldwin) [inline combat-trainer.lic]: oh for sure

Comment (drboldwin) [inline combat-trainer.lic]: can I just do, invoke(data['focus'], nil)?

**Pull request merged**